lib/kuzzle/kuzzle.ts

```diff
@@ -79,7 +82,56 @@ Reflect.defineProperty(global, 'kuzzle', {
  * @extends EventEmitter
  */
 class Kuzzle extends KuzzleEventEmitter {
-  constructor (config) {
+  public config: JSONObject;
+  public _state: kuzzleStateEnum = kuzzleStateEnum.STARTING;
+  public log: Logger;
+  public rootPath: string;
+  // Internal index bootstrapper and accessor
+  public internalIndex: InternalIndexHandler;
+
+  public pluginsManager: PluginsManager;
+  public tokenManager: TokenManager;
+  public passport: PassportWrapper;
+
+  // The funnel dispatches messages to API controllers
+  public funnel: Funnel;
+
+  // The router listens to client requests and pass them to the funnel
+  public router: Router;
+
+  // Statistics core component
+  public statistics: Statistics;
+
+  // Network entry point
+  public entryPoint: EntryPoint;
+
+  // Validation core component
+  public validation: Validation;
+  
+  // Dump generator
+  public dumpGenerator: DumpGenerator;
+
+  // Vault component (will be initialized after bootstrap)
```

Comment:
It's really a nitpicking but are the comments displayed by the IDE when they are with `//` and not  `/* */` ?

Reply:
Ah didn't know that